Applications/Spreadsheet/HelpWindow.cpp

```diff
@@ -97,18 +100,59 @@ HelpWindow::HelpWindow(GUI::Window* parent)
     m_listview->set_model(HelpListModel::create());
 
     m_webview = splitter.add<Web::OutOfProcessWebView>();
+    m_webview->on_link_click = [this](auto& url, auto&, auto&&) {
+        ASSERT(url.protocol() == "spreadsheet");
+        if (url.host() == "example") {
+            auto entry = LexicalPath(url.path()).basename();
+            auto doc_option = m_docs.get(entry);
+            if (doc_option.is_object()) {
+                auto& doc = doc_option.as_object();
+
+                auto example_data_value = doc.get_or("example_data", JsonObject {});
+                if (example_data_value.is_object()) {
+                    auto& example_data = example_data_value.as_object();
+                    const auto& name = url.fragment();
+                    if (auto value = example_data.get(name); value.is_object()) {
+                        auto dialog = GUI::Window::construct(this);
+                        dialog->resize(size());
+                        dialog->set_icon(icon());
+                        dialog->set_title(String::formatted("Spreadsheet Help - Example {} for {}", name, entry));
+                        auto& widget = dialog->set_main_widget<SpreadsheetWidget>(NonnullRefPtrVector<Sheet> {}, false);
+                        auto sheet = Sheet::from_json(value.as_object(), widget.workbook());
+                        if (!sheet) {
+                            GUI::MessageBox::show_error(this, String::formatted("Corrupted example '{}' in '{}'", name, url.path()));
+                        } else {
+                            widget.add_sheet(sheet.release_nonnull());
+                            dialog->show();
+                        }
+                    } else {
+                        GUI::MessageBox::show_error(this, String::formatted("Example '{}' not found for '{}'", name, url.path()));
+                    }
+                } else {
+                    GUI::MessageBox::show_error(this, String::formatted("No example data found for '{}'", url.path()));
+                }
+            } else {
+                GUI::MessageBox::show_error(this, String::formatted("No documentation entry found for '{}'", url.path()));
+            }
```

Comment:
There's an unpleasant amount of nesting going on here.
Can we get this code unnested (using early returns) and out of the constructor? :)

Reply:
Sometimes I miss out on the most obvious things...